### PR TITLE
machined: switch remaining Varlink overs over to use json_dispatch_pidref() and friends

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

**Comment by poettering** [comment]
This reworks the APIs recently added in #34681 to use the new concepts from #34703

File: src/basic/pidref.h [hunk 1]
```diff
@@ -13,6 +13,11 @@ typedef struct PidRef {
 
 #define PIDREF_NULL (const PidRef) { .fd = -EBADF }
 
+/* A special pidref value that we are using when a PID shall be automatically acquired from some surrounding
+ * context, for example connection peer. Much like PIDREF_NULL it will be considerd unset by
+ * pidref_is_set().*/
+#define PIDREF_AUTOMATIC (const PidRef) { .pid = PID_AUTOMATIC, .fd = -EBADF }
```
Comment (YHNdnzj) [inline src/basic/pidref.h:22]: Hmm, PID_AUTOMATIC is defined in process-util.h, but not included here.

Comment (poettering) [reply src/basic/pidref.h:22]: yeah, otherwise we'd have a cyclic dep... this is on purpose

File: src/machine/machine-varlink.c [hunk 2]
```diff
@@ -200,25 +200,25 @@ static int lookup_machine_by_name(sd_varlink *link, Manager *manager, const char
         return 0;
 }
 
-static int lookup_machine_by_pid(sd_varlink *link, Manager *manager, pid_t pid, Machine **ret_machine) {
-        _cleanup_(pidref_done) PidRef pidref = PIDREF_MAKE_FROM_PID(pid);
+static int lookup_machine_by_pid(sd_varlink *link, Manager *manager, const PidRef *pidref, Machine **ret_machine) {
```
Comment (yuwata) [inline src/machine/machine-varlink.c]: I'd like to rename this to `lookup_machine_by_pidref()`.

File: src/machine/machine-varlink.c [hunk 3]
```diff
@@ -228,7 +228,7 @@ static int lookup_machine_by_pid(sd_varlink *link, Manager *manager, pid_t pid,
         return 0;
 }
 
-int lookup_machine_by_name_or_pid(sd_varlink *link, Manager *manager, const char *machine_name, pid_t pid, Machine **ret_machine) {
+int lookup_machine_by_name_or_pid(sd_varlink *link, Manager *manager, const char *machine_name, const PidRef *pidref, Machine **ret_machine) {
```
Comment (yuwata) [inline src/machine/machine-varlink.c]: Also here. `_name_or_pidref()`.

File: src/shared/varlink-io.systemd.Machine.c [hunk 4]
```diff
@@ -10,7 +10,7 @@
         SD_VARLINK_FIELD_COMMENT("If non-null the name of a machine."),                                                                        \
         SD_VARLINK_DEFINE_INPUT(name, SD_VARLINK_STRING, SD_VARLINK_NULLABLE),                                                                 \
         SD_VARLINK_FIELD_COMMENT("If non-null the PID of a machine. Special value 0 means to take pid of the machine the caller is part of."), \
-        SD_VARLINK_DEFINE_INPUT(pid, SD_VARLINK_INT, SD_VARLINK_NULLABLE),                                                                     \
+        SD_VARLINK_DEFINE_INPUT_BY_TYPE(pid, ProcessId,  SD_VARLINK_NULLABLE),                                                                 \
```
Comment (yuwata) [inline src/shared/varlink-io.systemd.Machine.c]: doubled space.

File: src/basic/pidref.c [hunk 5]
```diff
@@ -428,6 +428,10 @@ int pidref_wait_for_terminate(const PidRef *pidref, siginfo_t *ret) {
         }
 }
 
+bool pidref_is_automatic(const PidRef *pidref) {
```
Comment (yuwata) [inline src/basic/pidref.c:431]: I'd like to split the first commit into at least two, pidref and json part, then machined part.

**Review by yuwata — CHANGES_REQUESTED** [review]

File: src/basic/pidref.h (hunk 1 above)
Comment (yuwata) [reply src/basic/pidref.h:22]: Then, please move this and pidref_is_automatic() to process-util.h .
process-util.h already has several pidref_is_xyz(). Hence, should be ok.

Comment (poettering) [reply src/basic/pidref.h:22]: i instead fixed the headers now to allow the cyclic dep for cases like this, similar how we do this elesewhere

**Comment by poettering** [comment]
Force pushed new version. PTAL.

File: src/basic/pidref.c [hunk 6]
```diff
@@ -428,6 +428,10 @@ int pidref_wait_for_terminate(const PidRef *pidref, siginfo_t *ret) {
...
+        return pidref && pid_is_automatic(pidref->pid);
+}
```
Comment (YHNdnzj) [inline src/basic/pidref.c:433]: This can be made static inline then?

**Review by yuwata — APPROVED** [review]
LGTM. But let's first merge #34769 and fix #34764.

**Pull request merged**